src/backend/metal/src/lib.rs

```diff
@@ -194,18 +193,31 @@ impl Instance {
                 panic!("window does not have a valid contentView");
             }
 
+            let existing: *mut Object = msg_send![view, layer];
             let class = Class::get("CAMetalLayer").unwrap();
```

Comment:
Unrelated but probably should use [`class!`](https://github.com/SSheldon/rust-objc/blob/master/src/lib.rs#L13) here instead

Reply:
good point! hopefully, this is only a one-time operation